chia/util/streamable.py

```diff
@@ -152,80 +155,121 @@ def streamable(cls: Any):
     """
 
     cls1 = strictdataclass(cls)
-    return type(cls.__name__, (cls1, Streamable), {})
+    t = type(cls.__name__, (cls1, Streamable), {})
+
+    parse_functions = []
+    try:
+        fields = cls1.__annotations__  # pylint: disable=no-member
+    except Exception:
+        fields = {}
+
+    for _, f_type in fields.items():
+        parse_functions.append(cls.function_to_parse_one_item(f_type))
```

Comment:
Is the intent here to automatically add parse functions for only the primitive types that are contained as members?

Reply:
The general idea is to traverse the `__annotations__` tree just once at class instantiation time, and put into a list the sub-parsing functions which do parsing and not introspection.